Jupyter/QuantBook.cs

```diff
@@ -368,6 +368,41 @@ public OptionHistory GetOptionHistory(Symbol symbol, DateTime date, Resolution?
             return new OptionHistory(_algorithm.HistoryProvider.GetHistory(requests.OrderByDescending(x => x.Symbol.SecurityType), _algorithm.TimeZone).Memoize());
         }
 
+        /// <summary>
+        /// Gets <see cref="FutureHistory"/> object for a given symbol, date and resolution
+        /// </summary>
+        /// <param name="symbol">The symbol to retrieve historical future data for</param>
+        /// <param name="date">Date of the data</param>
+        /// <param name="resolution">The resolution to request</param>
+        /// <returns>A <see cref="FutureHistory"/> object that contains historical future data.</returns>
+        public FutureHistory GetFutureHistory(Symbol symbol, DateTime date, Resolution? resolution = null)
+        {
+            SetStartDate(date.AddDays(1));
+            var future = _algorithm.Securities[symbol] as Future;
+
+            var provider = new BacktestingFutureChainProvider();
+            var allSymbols = provider.GetFutureContractList(future.Symbol, date);
+
+            var requests = future.ContractFilter.Filter(new FutureFilterUniverse(allSymbols, new Tick { Time = date }))
+                .Select(x =>
+                     new HistoryRequest(date.AddDays(-1),
+                                        date,
+                                        typeof(QuoteBar),
+                                        x,
+                                        resolution ?? future.Resolution,
+                                        future.Exchange.Hours,
+                                        MarketHoursDatabase.FromDataFolder().GetDataTimeZone(future.Symbol.ID.Market, future.Symbol, future.Type),
+                                        Resolution.Minute,
+                                        future.IsExtendedMarketHours,
+                                        future.IsCustomData(),
+                                        DataNormalizationMode.Raw,
+                                        LeanData.GetCommonTickTypeForCommonDataTypes(typeof(QuoteBar), future.Type))
```

Comment:
this method, `LeanData.GetCommonTickTypeForCommanDataTypes` will always return `TickType.Quote` when given `typeof(QuoteBar` -- any reason to call the method if we always want quotes?